src/main/resources/jupyter/google_sign_in.js

```diff
@@ -12,28 +12,46 @@ require.config({
 // TEMPLATED CODE
 // Leonardo has logic to find/replace templated values in the format $(...).
 // This will be replaced with the real Google Client ID before uploading to the notebook server.
-var clientId = $(googleClientId);
+var loginHint = $(googleLoginHint);
+var allowedOrigins = $(allowedOrigins);
```

Comment:
I'm thinking of templating `allowedOrigins` in from leo config. Initially it will contain a list of 2 values: the FC URL and the AoU URL.

It's just here for a security check so technically optional -- but I think it would be good to validate the source of the postMessage event against a known list.

Reply:
But then you're back to "DSP needs to update this list whenever someone else wants to build on top of Leo". We don't want other portal developers to have to ask to use our product! Flipside: what does validating the source of the postMessage get us? If the caller is sending us a valid token, why do we care where they're coming from? Maybe someone built their own UI to simplify some of their tasks (this has happened before...)

Reply:
Yeah.. point taken. I'm just nervous because every `postMessage` tutorial I've read says to always check the origin of the message. Otherwise malicious sites could send messages to your application and cause you to execute code. However we're also checking that `event.source == window.opener` so I guess that should be sufficient. I'll scrap the allowedOrigins.